File: models/produto_peca_model.dart
import 'package:gpp/src/models/pecas_model/peca_model.dart';
import 'package:gpp/src/models/produto/produto_model.dart';

class ProdutoPecaModel {
  int? idProdutoPeca;
  int? quantidadePorProduto;
  PecasModel? peca;
  ProdutoModel? produto;
  int? id_produto;
  int? id_peca;
  int? status;
  int? situacao;

  ProdutoPecaModel(
      {this.idProdutoPeca,
      this.quantidadePorProduto,
      this.peca,
      this.produto,
      this.id_produto,
      this.id_peca,
      this.situacao,
      this.status});

  factory ProdutoPecaModel.fromJson(Map<String, dynamic> json) {
    return ProdutoPecaModel(
        idProdutoPeca: json['id_produto_peca'],
        quantidadePorProduto: json['quantidade_por_produto'],
        id_produto: json['id_produto'],
        peca: json['peca'] == null ? null : PecasModel.fromJson(json['peca']),
        produto: json['produto'] == null ? null : ProdutoModel.fromJson(json['produto']),
        id_peca: json['id_peca'],
        situacao: json['situacao'],
        status: json['status']);
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = Map<String, dynamic>();
    data['quantidade_por_produto'] = quantidadePorProduto != null ? quantidadePorProduto : null;
    data['pecas'] = peca != null ? peca!.toJson() : null;

    data['id_produto_peca'] = this.idProdutoPeca;
    data['id_produto'] = this.id_produto;

    data['id_peca'] = this.id_peca;

    return data;
  }
}

File: gpp-api/src/main/java/com/develop/gpp/domain/entity/ProdutoPecaModel.java
package com.develop.gpp.domain.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
public class ProdutoPecaModel {

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long idProdutoPeca;

  private int quantidadePorProduto;

  // @ManyToOne
  // @JoinColumn(name = "id_peca")
  // private Peca peca;

  // @ManyToOne
  // @JoinColumn(name = "id_produto")
  // private Produto produto;

  private int status;

  private int situacao;

  // getters and setters
}
